src-gui/src/renderer/components/pages/swap/swap/init/redeemAddress/ReceiveAddressSelector.tsx

```diff
@@ -0,0 +1,57 @@
+import { Autocomplete, Box, TextField } from "@mui/material";
+import { useState, useEffect } from "react";
+import { getMoneroAddresses } from "renderer/rpc";
+
+export default function ReceiveAddressSelector({
+  onChange,
+  value,
+  isValidAddress,
+}: {
+  onChange: (address: string) => void;
+  value: string;
+  isValidAddress: boolean;
+}) {
+  const showError = value && !isValidAddress;
+  const [addresses, setAddresses] = useState<string[]>([]);
+
+  useEffect(() => {
```

Comment:
We can re-run this ever 5s or so. It's a cheap operation (single SQL query). If the component is rendered before the context is initialised, it'll fail and won't retry (which is not good)

Reply:
But we also want to use the older InitPage but we can still implement that logic there.